distributed/client.py

```diff
@@ -1827,8 +1827,10 @@ def list_datasets(self, **kwargs):
         return self.sync(self.scheduler.publish_list, **kwargs)
 
     @gen.coroutine
-    def _get_dataset(self, name):
-        out = yield self.scheduler.publish_get(name=name, client=self.id)
+    def _get_dataset(self, name, **kwargs):
+        out = yield self.scheduler.publish_get(
```

Comment:
If the only purpose is to sanitise the stack-trace, then a check on the dataset's existence could be made here before attempting to get, raising KeyError?

Reply:
So then raise here as well for backcompat? Wouldn't that still make things noisy?

Reply:
I was thinking raise here only, and since this is a local exception from the client, it will not get deeply buried.

Reply:
Ok that could work then. Never totally sure how things like Tornado deal with these issues. Plus not super familiar with what happens in this part of the codebase. 😄

Though maybe we do away with the check entirely and just call `get`. If it doesn't exist, it will return `None`, which is easy to check.

Reply:
Both sounds equally uncomplicated to me (checking before calling get, versus using `None` for not found), on an API that is not user-facing, so it's just a question of style.

Reply:
Thanks for the advice. PR ( https://github.com/dask/distributed/pull/1888 ) should do the trick. Please let me know what you think. :)